### ISSUE TITLE
convert snake_case identifiers to camelCase

### ISSUE THREAD

**Issue by mhuyck** [greenelab/adage-server#272]
Codeclimate noted several places in `experimentDetail.view.js` (during review of https://github.com/greenelab/adage-server/pull/269) where we use non-standard identifiers (`item_type`, `related_items` and `related_items`). These should be updated throughout the interface code to follow camelCase convention.

This needs to be handled as a separate issue because these identifiers (and others) are in use in many places, including in the back-end at `api.py`. The JSON response to a `search` produces a list of objects with the attributes above, which is why this Python identifier standard has "leaked" into the interface.

This should be addressed by ensuring that our JSON identifiers adhere to the camelCase standard wherever possible.

**Comment by mhuyck** [comment]
Digging into this a bit more and I'm going to revise my strategy a little. Specifically, where I said:

> This should be addressed by ensuring that our JSON identifiers adhere to the camelCase standard wherever possible.

I no longer think it's a great idea to go changing the (Python-based) api code to conform to JavaScript naming conventions. For one thing, many field names are passed through to us directly from the Django models via Tastypie's very helpful auto-mapping. It would be just as problematic to have camelCase naming appear in Python. Second, the choice of JavaScript for accessing the REST API is one made by the `interface` codebase and responsibility for making any adjustments should rest (hah.. no pun intended) with the interface code. I think this principle should be dubbed the *when in Rome* strategy. That is: when in Python, do as the Pythonistas do; when in JavaScript, do as the JS ninjas do.

Fortunately, we can do this identifier conversion right at the point where we define the AngularJS `$resource` using the technique I just found in debugging PR #292: there is a `transformResponse` callback specifically designed for modifying the response from a REST `$resource` and doing what we want. That technique was very helpful in #292, so I will apply it here as well to avoid modifying the Python to conform to our JavaScript standards.

**Comment by rzelayafavila** [comment]
@mhuyck: I agree with this, I was just thinking as well that the way the REST API does things should be agnostic to whatever is accessing the REST API (whether it is a JavaScript application, a Python package that is accessing it programmatically, or something else). What about (just for those parameters, which need to be passed to the API in snake_case) just making those parameters strings (like 'item_type' and 'related_items')? That way, it would be more clear that they are parameters to the request, not variables themselves. I believe that linters and code-climate would also stop complaining. What do you think?

**Comment by mhuyck** [comment]
That is a really good thought, @rzelayafavila, and may be the most pragmatic option.

I've actually found, as I work through each identifier, that many of them are just isolated to the front-end. These all seem to appear in older code, which I probably wrote in the time before we started using ESLint and adopting coding standards. I'm almost done cleaning those up and then will see which approach would produce the most clear results for those identifiers that cross the divide between the back-end and front-end.

**Comment by mhuyck** [comment]
It turned out to be easier to do the conversion from `item_type` to `itemType` (and two other snake_case identifiers) at the `transformResponse` level, so that is how I implemented it. I think the resulting code is pretty easy to follow... PR coming shortly in which you can let me know if you agree.
